### PR TITLE
86cuex00r/UI/user-listing

### PR DESCRIPTION
**Pull request by amaljoeqb**
This PR addresses the task of designing users table for listing the users in admin panel.

Changes:

- Added package @mui/x-data-grid
- Use datagrid to develop users table
- Added sampleUsers.json for testing purposes
- Added type for user

image

### CODE REVIEW

File: src/pages/UserListing/UsersListingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { UsersListingPageWrapper } from './UsersListingPage.styled';
+import { UsersTable } from './components/Table/UsersTable';
+import users from '../../data/sampleUsers.json';
+
+export const UsersListingPage = () => {
+    return (
+        <UsersListingPageWrapper>
```
Comment (akshay-krishna) [inline src/pages/UserListing/UsersListingPage.tsx]: Can you try this:
<UsersListingPageWrapper component={'section'}>

File: src/pages/UserListing/components/Table/UsersTable.styled.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { styled } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+
+export const UsersTableWrapper = styled(Box)(() => ({}));
```
Comment (RJR-003) [inline src/pages/UserListing/components/Table/UsersTable.styled.ts]: Do we really need this wrapper if it is not doing any styling?

**Review by RJR-003 — COMMENTED** [review]
Hi i have put a comment.

File: src/pages/UserListing/UsersListingPage.tsx (hunk 1 above)
Comment (amaljoeqb) [reply src/pages/UserListing/UsersListingPage.tsx]: done

File: src/pages/UserListing/components/Table/UsersTable.styled.ts (hunk 2 above)
Comment (amaljoeqb) [reply src/pages/UserListing/components/Table/UsersTable.styled.ts]: removed

File: src/interfaces/user.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface User {
```
Comment (akshay-krishna) [inline src/interfaces/user.d.ts]: better to make interface as IUser

**Review by RJR-003 — APPROVED** [review]

**Review by vipinmsqburst — APPROVED** [review]

**Pull request merged**